controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -175,9 +175,39 @@ func (r *OpenshiftAssistedControlPlaneReconciler) Reconcile(ctx context.Context,
 		log.Error(err, "failed to get k8s version from release image")
 	}
 	oacp.Status.Version = k8sVersion
+	upgradeRequested := determineUpgrade(ctx, oacp)
+	if upgradeRequested {
+		//TODO: Handle upgrade request
+	}
 	return ctrl.Result{}, r.reconcileReplicas(ctx, oacp, cluster)
 }
 
+func determineUpgrade(ctx context.Context, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) bool {
```

Comment:
should we rename the method to `isUpgradeRequested`? We could then do
```
if isUpgradeRequested(ctx, oacp) {
  // handle upgrade
}
```

Reply:
Good idea!